assets/js/listener.js

```diff
@@ -6,40 +6,50 @@ const commentTextArea = document.getElementById('comment');
 const thankYouText = document.getElementById('thank-you');
 const nav = document.getElementById('site-nav');
 const versionPanel = document.getElementById('version-panel');
-document.addEventListener('DOMContentLoaded', updateTextArea);
 
+// Single click event listener for the entire document
 document.addEventListener('click', function(event) {
     const { target } = event;
-    if (target.matches('.feedback-issue')) {
-        gtag('event', 'submit_issue_click');
-    }
-    else if (target.matches('.feedback-edit')) {
-        gtag('event', 'edit_page_click');
-    }
-    else if (target.matches('.feedback-forum')) {
-        gtag('event', 'forum_link_click');
-    }
-    else if (target.matches('.feedback-button')) {
-        sendButton.disabled = false;
-    }
-    else if (target.matches('.send-button')) {
-        sendFeedback();
-    }
-    else if (target.matches('.copy-button')) {
+
+    // Handle old-style buttons first
+    if (target.matches('.copy-button') && target.hasAttribute('data-text')) {
         window.navigator.clipboard.writeText(target.getAttribute('data-text'));
+        return; // Exit early to avoid multiple handlers
     }
-});
 
-nav.addEventListener('scroll',(e)=>{  
-    if(nav.scrollTop > 0){
-      versionPanel.classList.add("nav-shadow");
-    }else{
-      versionPanel.classList.remove("nav-shadow");
+    // Handle new-style buttons and other clicks
+    const clickHandlers = {
+        '.feedback-issue': () => gtag('event', 'submit_issue_click'),
+        '.feedback-edit': () => gtag('event', 'edit_page_click'),
+        '.feedback-forum': () => gtag('event', 'forum_link_click'),
+        '.feedback-button': () => sendButton.disabled = false,
+        '.send-button': () => sendFeedback(),
+        '.tab-button': () => switchTab(event, target.getAttribute('data-tab')),
+        '.copy-code-button': () => copyCode(target),
+        '.copy-curl-button': () => copyAsCurl(target)
+    };
+
+    for (const [selector, handler] of Object.entries(clickHandlers)) {
+        if (target.matches(selector)) {
+            handler();
+            break;
+        }
     }
-  });
+});
+
+// Event listeners
+document.addEventListener('DOMContentLoaded', updateTextArea);
 
 commentTextArea.addEventListener('input', updateTextArea);
 
+nav.addEventListener('scroll', (e) => {
+    if (nav.scrollTop > 0) {
+        versionPanel.classList.add('nav-shadow');
+    } else {
+        versionPanel.classList.remove('nav-shadow');
+    }
+});
```

Comment:
This needs debouncing; in its current shape it could drastically impact performance.